guides/git-pr-review.md

```diff
@@ -0,0 +1,105 @@
+
+
+### PR Review Steps
+
+#### 1. Review the PR and Issue
+
+Read through the PR to see what was done and the discussion people have had on
+there. Confirm the GitHub "fixes" or "closes" magic words are used to
+automatically close the related issues.
+
+Then, if there is a corresponding issue or issues, read through the original
+content of the issue and then read through the discussion and clarification.
+
+Many times there are going to be other "requirements" or clarifications that 
+might influence the PR beyond the original issue text.
+
+If the PR does not appear to fix/implement the issue or the description is not
+clear in the PR text, then stop here and ask questions. Document any new
+understanding or thoughts in the PR for future reference.
+
+#### 2. Review the code
+
+Look through the code and see if it appears to answer the question laid out by
+the PR and Issue text. It should answer only the issue and not other bugs etc.
+Other ones should be a in a separate PR (which will make it easier for review,
+and bisecting if a bug is added and found later on). 
+
+The code should also be reviewed for style, maintainability, and to make sure
+it fits into the framework of the code in the GitHub repository. It would be
+better to maintain similarity to current code unless there is a need to make
+changes to the repository code framework.  
+
+Python code should follow the [Python style
+guide](https://github.com/spacetelescope/style-guides/blob/master/guides/python.md).
+Also, automated PEP 8 style checking should happen as part of the 
+[CI / automated test infrastructure](https://github.com/spacetelescope/style-guides/issues/73).
+
+If the code appears to have bugs or implementation issues, then stop now and
+ask for clarification from the author (and maybe get them to make code
+changes).
+
+#### 3. Check and Run the tests
+
+1. Confirm the tests are running and passing in the CI. Be mindful that green
+   check marks can be false positives, so read the logs to be sure before
+   merge. Did you check the jobs that are allowed to fail? Are they failing for
+   the right things?
+
+2. Are other tests needed based on new functionality OR to check for the bug
+   from the issue. 
+
+If the tests do not run or fail, well, definitely stop now and get the author
+to fix the code so the tests pass.  
+
+If more tests are needed, request them from the author.
+
+#### 4. Review the documentation
+
+Read through the relevant portion of the documentation to see if it is
+sufficient or correct given the changes in the PR.  Does more documentation
+need to be added? Does the existing documentation need to be modified based on
+the PR? Can I come back in 6 months and understand how things work from this
+documentation? Can a new user or maintainer understand it?
+
+Is a change log entry necessary? If yes, was it included in the right place?
+
+
+#### 5. Run the branch locally
+
+Fetch the branch from GitHub and run the tests locally. If there are GUI
```

Comment:
I think this requirement is a bit heavy handed, so I'm suggesting the below as a lighter-weight alternative.  I agree that the GUI case it's worth emphasizing, though!

```suggestion

Local testing is not always necessary if you are confident that the changes the PR introduced are thoroughly tested via the CI tests (either via new tests in the PR or existing tests). However, some features, like GUI elements, are not easily tested via CI, or CI may not yet be set up in the package you are working on.  In that case you must  run tests locally. To do this you will need to fetch the branch from GitHub, merge it with master in a local testing branch, and run the tests (manual or automatically) yourself. 
If there are GUI
```

Reply:
I made the change. 

But, I will say that there have been several times that some developers work in an unclean environment with several pieces of work happening at the same time (not using branches quite right) and so when their branch is brought down to a clean environment things don't work. So, I'll accept your wording, and this is a guide, but I think there is a lot to be said for testing locally.